README
Java GUI configurator for TANGO HDB++

The TANGO archiving system is a tool allowing TANGO users to store the readings coming from a TANGO based control system into a database.
The archived data are essential for the day by day operation of complex scientific facilities for instance.
They can be used for long term monitoring of subsystems, statistics, parameters correlation or comparison of operating setups over time.
To take advantage of the fast and lightweight event-driven communication provided by TANGO release 8 with the adoption of ZeroMQ,
	a novel archiving system for the TANGO Controls framework, named HDB++, has been designed and developed,
	resulting from a collaboration between Elettra and ESRF at the beginning.
HDB++ design allows TANGO users to store data with microsecond timestamp resolution into traditional database management systems
	such as MySQL or into NoSQL databases such as Apache Cassandra. 


See:
http://www.esrf.fr/computing/cs/tango/tango_doc/tools_doc/hdb++-configurator/


CHANGELOG.md
# Changelog

#### hdb_configurator-3.26 - 02/10/20:
    Remove contexts ALWAYS and NEVER on context selection to set attribute

#### hdb_configurator-3.25 - 12/06/20:
    Another statistics display improvement
    Add methods to be used from outside

#### hdb_configurator-3.24 - 03/06/20:
    Fix a bug in wildcard (if event from another tango host)
    Another statistics display improvement

#### hdb_configurator-3.23 - 28/05/20:
    Fix small bug and improve statistics display.

#### hdb_configurator-3.22 - 20/02/20:
    Add setReorderingAllowed(false) for all tables.

#### hdb_configurator-3.21 - 13/01/20:
    Small features for extenal process

#### hdb_configurator-3.20 - 07/01/20:
    Improve subscriber and attribute selection on archived attributes.

#### hdb_configurator-3.19 - 20/12/19:
    Add a context manager button if specified.

#### hdb_configurator-3.17 - 08/10/19:
    Update ScalarTableViewer tool

#### hdb_configurator-3.16 - 04/10/19:
    Add a subscriber selector to frequency trend.
    Add a menu item to set visible the statistics chart

#### hdb_configurator-3.15 - 25/09/19:
    Add a chart in statistics dialog

#### hdb_configurator-3.14 - 18/09/19:
    Fix a problem in distribution display.

#### hdb_configurator-3.13 - 17/09/19:
    Cosmetics

#### hdb_configurator-3.12 - 02/08/19:
    Add an attribute selection using a device wildcard

#### hdb_configurator-3.11 - 14/03/19:
    Fix a bug in case of TANGO_HOST is an alias

#### hdb_configurator-3.10 - 08/01/19:
    Manage context NEVER if exist.

#### hdb_configurator-3.9 - 06/11/18:
    Add a HDB++/LimitedAccessTangoHosts free property
    Small features for extenal process

#### hdb_configurator-3.8 - 16/10/18:
    Fix a problem on diagnostics frame if a subscriber is down.

#### hdb_configurator-3.5 - 28/05/18:
    Fix a problem in double click on Diagnostics Frequency column.
    Subscribe event is now stateless.

#### hdb_configurator-3.4 - 04/05/18:
    Improve ServerInfoTable management.

#### hdb_configurator-3.3 - 25/04/18:
    Improve attribute selection in table.

#### hdb_configurator-3.2 - 25/04/18:
    TTL attribute table added.

#### hdb_configurator-3.1 - 15/03/18:
    TTL tested in real control system.
    Problem on No Heartbeat display fixed.

#### hdb_configurator-3.0 - 25/09/17:
    TTL management added.

#### hdb_configurator-2.2 - 23/05/17:
    Moved to gitub.

#### hdb_configurator-2.1 - 28/02/17:
    First release installed with strategies

#### hdb_configurator-2.0a - 14/12/16:
    Moved to maven project

#### hdb_configurator-2.0 - 02/11/16:
    Context management added.

#### hdb_configurator-1.5a - 21/09/16:
    Change date format in daily saving.

#### hdb_configurator-1.5 - 05/09/16:
    Server information table added.
    Add global statistics on distribution frame.
    Fix a small problem when sort statistics
    Add a copy attributeList name in statistics window.

#### hdb_configurator-1.4 - 24/08/16:
    Add a litle tool to manage Stat/Stop all attributes on specified subscribers.
    Fix a small problem in distribution display.
    Add a copy error message added in diagnostics.

#### hdb_configurator-1.3 - 06/06/16:
    Fix a problem on default TANGO_HOST

#### hdb_configurator-1.2 - 27/05/16:
    Fix a problem for DevFailed/NamedDevFailedList exception

#### hdb_configurator-1.1 - 13/05/16:
    Fix problem of refreshing in attributeList tree.

#### hdb_configurator-1.0 - 12/05/16:
    Change package name (reset revision number)

#### jhdbcpp-2.7 - 03/05/16:
    few methods added for external class.

#### jhdbcpp-2.6 - 19/12/15:
    Remove event tester launch in StatisticsDialog menu (does not work fine).

#### jhdbcpp-2.5 - 16/12/15:
    Fix a problem when moving paused attributes

#### jhdbcpp-2.4 - 26/11/15:
    Refactor to remove cross links.

#### jhdbcpp-2.3 - 26/11/15:
    Interface with jhdbvewer to read attributeList in hdb.

#### jhdbcpp-2.2 - 24/09/15:
    Fix a little bug on plain/bold attributeList display

#### jhdbcpp-2.1 - 19/05/15:
    Subscriber alias management added.
    Performance view added in distribution
    Compatibility with Java-7

#### jhdbcpp-2.0 - 23/04/15:
    Change GUI look and feel.
    Add pause attributeList state.
    Attribute lists are updated from events

#### jhdbcpp-1.13 - 08/04/15:
    small chamges.

#### jhdbcpp-1.12 - 03/03/15:
    Info on attributeList failure frequency and on E.S. manager added.
    Trend on HDB storage frequency added.

#### jhdbcpp-1.11 - 06/02/15:
    Add a table to view all attribues on error.

#### jhdbcpp-1.10 - 20/01/15:
    Table added to distribution dialog

#### jhdbcpp-1.9 - 16/01/15:
    Events number added to distribution chart.

#### jhdbcpp-1.8 - 19/12/14:
    Improve attributeList error window.
    Add a selector to change TANGO_HOST
    Event tester call added.

#### jhdbcpp-1.7 - 06/11/14:
    FQDN event subscriber name to be added to manager.
    Events since reset added.
    Save statistics added.

#### jhdbcpp-1.6 - 30/10/14:
    Display event frequency added.

#### jhdbcpp-1.5 - 13/10/14:
    Display error message on failed attributeList list.
    Display HDB name if property set.

#### jhdbcpp-1.4 - 24/09/14:
    Diag viewers are now in a JTable object.
    A double click on table cell displays attributeList list.

#### jhdbcpp-1.3 - 22/09/14:
    Fix a bug in archiver device creation

#### jhdbcpp-1.2 - 05/09/14:
    Add a graph chart to display distribution.

#### jhdbcpp-1.1 - 28/08/14:
    Add/Remove subscriber added.

#### jhdbcpp-1.0 - 06/08/14:
    Initial Revision


src/main/java/org/tango/hdb_configurator/common/ReleaseNote.java

package org.tango.hdb_configurator.common;

/**
 * HTML code to display Release Notes for this package.
 * It is generated by Pogo classes from a text file.
 */

public interface ReleaseNote {
	String str = 
		"<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 3.2//EN\">\n" + 
		"<html>\n" + 
		"<head>\n" + 
		"<title> Release Note </title>\n" + 
		"</head>\n" + 
		"<body text=\"#000000\" bgColor=\"#FFFFFF\" link=\"#0000FF\" vLink=\"#7F00FF\" aLink=\"#FF0000\">\n" + 
		"<p><!-------TITLE------></p>\n" + 
		"<center>	<h2>org.tango.hdb_configurator.tools Release Notes </h2>\n" + 
		"	(Generated 23-05-2017  13:18:05)</center><br>\n" + 
		"<li><b>hdb_configurator-2.2 - 23/05/17:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Moved to gitub.<br>" + 
		"<li><b>hdb_configurator-2.1 - 28/02/17:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; First release installed with strategies<br>" + 
		"<li><b>hdb_configurator-2.0a - 14/12/16:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Moved to maven project<br>" + 
		"<li><b>hdb_configurator-2.0 - 02/11/16:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Context management added.<br>" + 
		"<li><b>hdb_configurator-1.5a - 21/09/16:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Change date format in daily saving.<br>" + 
		"<li><b>hdb_configurator-1.5 - 05/09/16:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Server information table added.<br>" + 
		"&nbsp; &nbsp; &nbsp; Add global statistics on distribution frame.<br>" + 
		"&nbsp; &nbsp; &nbsp; Fix a small problem when sort statistics<br>" + 
		"&nbsp; &nbsp; &nbsp; Add a copy attributeList name in statistics window.<br>" + 
		"<li><b>hdb_configurator-1.4 - 24/08/16:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Add a litle tool to manage Stat/Stop all attributes on specified subscribers.<br>" + 
		"&nbsp; &nbsp; &nbsp; Fix a small problem in distribution display.<br>" + 
		"&nbsp; &nbsp; &nbsp; Add a copy error message added in diagnostics.<br>" + 
		"<li><b>hdb_configurator-1.3 - 06/06/16:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Fix a problem on default TANGO_HOST<br>" + 
		"<li><b>hdb_configurator-1.2 - 27/05/16:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Fix a problem for DevFailed/NamedDevFailedList exception<br>" + 
		"<li><b>hdb_configurator-1.1 - 13/05/16:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Fix problem of refreshing in attributeList tree.<br>" + 
		"<li><b>hdb_configurator-1.0 - 12/05/16:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Change package name (reset revision number)<br>" + 
		"<li><b>jhdbcpp-2.7 - 03/05/16:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; few methods added for external class.<br>" + 
		"<li><b>jhdbcpp-2.6 - 19/12/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Remove event tester launch in StatisticsDialog menu (does not work fine).<br>" + 
		"<li><b>jhdbcpp-2.5 - 16/12/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Fix a problem when moving paused attributes<br>" + 
		"<li><b>jhdbcpp-2.4 - 26/11/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Refactor to remove cross links.<br>" + 
		"<li><b>jhdbcpp-2.3 - 26/11/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Interface with jhdbvewer to read attributeList in hdb.<br>" + 
		"<li><b>jhdbcpp-2.2 - 24/09/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Fix a little bug on plain/bold attributeList display<br>" + 
		"<li><b>jhdbcpp-2.1 - 19/05/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Subscriber alias management added.<br>" + 
		"&nbsp; &nbsp; &nbsp; Performance view added in distribution<br>" + 
		"&nbsp; &nbsp; &nbsp; Compatibility with Java-7<br>" + 
		"<li><b>jhdbcpp-2.0 - 23/04/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Change GUI look and feel.<br>" + 
		"&nbsp; &nbsp; &nbsp; Add pause attributeList state.<br>" + 
		"&nbsp; &nbsp; &nbsp; Attribute lists are updated from events<br>" + 
		"<li><b>jhdbcpp-1.13 - 08/04/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; small chamges.<br>" + 
		"<li><b>jhdbcpp-1.12 - 03/03/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Info on attributeList failure frequency and on E.S. manager added.<br>" + 
		"&nbsp; &nbsp; &nbsp; Trend on HDB storage frequency added.<br>" + 
		"<li><b>jhdbcpp-1.11 - 06/02/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Add a table to view all attribues on error.<br>" + 
		"<li><b>jhdbcpp-1.10 - 20/01/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Table added to distribution dialog<br>" + 
		"<li><b>jhdbcpp-1.9 - 16/01/15:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Events number added to distribution chart.<br>" + 
		"<li><b>jhdbcpp-1.8 - 19/12/14:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Improve attributeList error window.<br>" + 
		"&nbsp; &nbsp; &nbsp; Add a selector to change TANGO_HOST<br>" + 
		"&nbsp; &nbsp; &nbsp; Event tester call added.<br>" + 
		"<li><b>jhdbcpp-1.7 - 06/11/14:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; FQDN event subscriber name to be added to manager.<br>" + 
		"&nbsp; &nbsp; &nbsp; Events since reset added.<br>" + 
		"&nbsp; &nbsp; &nbsp; Save statistics added.<br>" + 
		"<li><b>jhdbcpp-1.6 - 30/10/14:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Display event frequency added.<br>" + 
		"<li><b>jhdbcpp-1.5 - 13/10/14:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Display error message on failed attributeList list.<br>" + 
		"&nbsp; &nbsp; &nbsp; Display HDB name if property set.<br>" + 
		"<li><b>jhdbcpp-1.4 - 24/09/14:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Diag viewers are now in a JTable object.<br>" + 
		"&nbsp; &nbsp; &nbsp; A double click on table cell displays attributeList list.<br>" + 
		"<li><b>jhdbcpp-1.3 - 22/09/14:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Fix a bug in archiver device creation<br>" + 
		"<li><b>jhdbcpp-1.2 - 05/09/14:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Add a graph chart to display distribution.<br>" + 
		"<li><b>jhdbcpp-1.1 - 28/08/14:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Add/Remove subscriber added.<br>" + 
		"<li><b>jhdbcpp-1.0 - 06/08/14:</b><br>" + 
		"&nbsp; &nbsp; &nbsp; Initial Revision<br>" + 
		"</body>\n" + 
		"</html>\n"
	;
}